<?xml version="1.0"?>
<launch>
  <node pkg="nodelet" type="nodelet" name="collision_avoidance_manager" args="manager" output="screen" />


  <!-- Pass in collision_avoidance_config_file:=/path/to/config.yaml to change options. -->
  <arg name="collision_avoidance_config_file" default="$(find collision_avoidance)/config/collision_avoidance_uav.yaml" />

  <node pkg="nodelet" type="nodelet" name="collision_avoidance_nodelet" args="load collision_avoidance/CollisionAvoidanceNodelet collision_avoidance_manager" output="screen">
    <rosparam file="$(arg collision_avoidance_config_file)" command="load" />

    <remap from="cloud" to="/octomap_point_cloud_centers" />
    <remap from="cloud_1" to="/camera/depth/points" />
    <remap from="odometry" to="/mavros/local_position/odom" />
    <remap from="~/control" to="/mavros/setpoint_velocity/cmd_vel" />
    <remap from="~/path" to="~/path" />
    <remap from="~/obstacles" to="~/obstacles" />
    <remap from="move_to" to="move_to2" />

    <param name="num_cloud_topics" value="2" />

    <param name="robot_frame_id" value="base_stabilized_link" />

    <param name="distance_converged" value="0.4" />
    <param name="yaw_converged" value="0.5" />

    <param name="frequency" value="20.0" />

    <param name="max_xy_vel" value="1.0" />
    <param name="max_z_vel" value="1.0" />
    <param name="max_yaw_rate" value="40.0" />
    <param name="h_m" value="1.0" />

    <param name="radius" value="0.5" />
    <param name="height" value="0.2" />
    <param name="min_distance_hold" value="1.0" />

    <param name="security_distance" value="1.0" />
    <param name="epsilon" value="0.1" />

    <param name="polar_size" value="360" />

    <param name="max_times_backwards" value="10" />
    <param name="max_times_backwards_total" value="1000" />
    <param name="max_direction_change" value="60" />

    <param name="look_ahead_distance" value="4.0" />
    <param name="look_forward" value="true" />
    <param name="move_while_yawing" value="false" />
  </node>

  <!-- <node pkg="nodelet" type="nodelet" name="laserscan_to_pointcloud_nodelet" args="load laserscan_to_pointcloud/LTP collision_avoidance_manager" output="screen">
    <param name="scan_topic" type="str" value="scan" />
    <param name="cloud_topic" type="str" value="cloud" />
  </node> -->

</launch>
